File: ce/snippets/visualbasic/CRMV8/reportingvb/vb/getreporthistorylimit1.vb
' <snippetgetreporthistorylimit1>



 ' Query for an existing report: Account Overview. This is a default report in Microsoft Dynamics CRM.                    
 Dim reportQuery As New QueryByAttribute(Report.EntityLogicalName)
 reportQuery.AddAttributeValue("name", "Account Overview")
 reportQuery.ColumnSet = New ColumnSet("reportid")

 ' Get the report.
 Dim retrieveReports As EntityCollection = _serviceProxy.RetrieveMultiple(reportQuery)

 ' Convert retrieved Entity to a report
 Dim retrievedReport As Report = CType(retrieveReports.Entities(0), Report)
 Console.WriteLine("Retrieved the 'Account Overview' report.")

 ' Use the Download Report Definition message.
 Dim reportHistoryRequest As GetReportHistoryLimitRequest =
  New GetReportHistoryLimitRequest With {
  .ReportId = retrievedReport.ReportId.Value
 }

 Dim reportHistoryResponse As GetReportHistoryLimitResponse =
  CType(_serviceProxy.Execute(reportHistoryRequest), GetReportHistoryLimitResponse)

 ' Access the history limit data
 Dim historyLimit As Integer = reportHistoryResponse.HistoryLimit

 Console.WriteLine("The report history limit is {0}.", historyLimit)

' </snippetgetreporthistorylimit1>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/reporting/cs/getreporthistorylimit1.cs
// <snippetgetreporthistorylimit1>



 // Query for an existing report: Account Overview. This is a default report in Microsoft Dynamics CRM.				    
 QueryByAttribute reportQuery = new QueryByAttribute(Report.EntityLogicalName);
 reportQuery.AddAttributeValue("name", "Account Overview");
 reportQuery.ColumnSet = new ColumnSet("reportid");

 // Get the report.
 EntityCollection retrieveReports = _serviceProxy.RetrieveMultiple(reportQuery);

 // Convert retrieved Entity to a report
 Report retrievedReport = (Report)retrieveReports.Entities[0];
 Console.WriteLine("Retrieved the 'Account Overview' report.");

 // Use the Download Report Definition message.
 GetReportHistoryLimitRequest reportHistoryRequest = new GetReportHistoryLimitRequest
 {
     ReportId = retrievedReport.ReportId.Value
 };

 GetReportHistoryLimitResponse reportHistoryResponse = (GetReportHistoryLimitResponse)_serviceProxy.Execute(reportHistoryRequest);

 // Access the history limit data
 int historyLimit = reportHistoryResponse.HistoryLimit;

 Console.WriteLine("The report history limit is {0}.", historyLimit);

// </snippetgetreporthistorylimit1>